packages/cozy-sharing/src/state.js

```diff
@@ -299,6 +310,43 @@ export const getRecipients = (state, docId) => {
   return recipients
 }
 
+export const getRecipientsWithGroups = (sharings, docId) => {
+  const recipients = sharings.flatMap(sharing => {
+    const type = getDocumentSharingType(sharing, docId)
+    const sharingId = sharing.id
+
+    const members = sharing.attributes.members.map((m, idx) => ({
+      ...m,
+      type: m.read_only ? 'one-way' : type,
+      sharingId,
+      index: idx,
+      avatarPath: `/sharings/${sharingId}/recipients/${idx}/avatar`
+    }))
+
+    const groups =
+      sharing.attributes.groups
+        ?.map((g, idx) => ({
+          ...g,
+          sharingId,
+          index: idx,
```

Comment:
Seems like this is used to fill React components' index, so it seems to be important to have unique index. So I suppose this need to be fixed before approving the PR 😄

Reply:
This is deliberate because it allows you to revoke a specific group or member. The stack request takes the index of the array to find out which one it needs to modify (cf: [doc](https://docs.cozy.io/en/cozy-stack/sharing/#delete-sharingssharing-idrecipientsindex))

However, I'm going to check that this index is not used in the display of react components.

Reply:
I've added an index by group and by member for the same share to manage revocations. I'm taking advantage of this to remove the global index from the recipients to use objects directly from the map when the list is displayed. It's probably not optimal but given that we don't have an id from the data it's the best solution I can think of